node/src/components/block_proposer/deploy_sets.rs

```diff
@@ -50,16 +71,33 @@ impl BlockProposerDeploySets {
     /// Prunes expired deploy information from the BlockProposerState, returns the total deploys
     /// pruned
     pub(crate) fn prune(&mut self, current_instant: Timestamp) -> usize {
-        let pending = prune_deploys(&mut self.pending, current_instant);
+        let pending = prune_pending_deploys(&mut self.pending, current_instant);
         let finalized = prune_deploys(&mut self.finalized_deploys, current_instant);
         pending + finalized
     }
 }
 
-/// Prunes expired deploy information from an individual DeployCollection, returns the total
+/// Prunes expired deploy information from an individual deploy collection, returns the total
 /// deploys pruned
-pub(super) fn prune_deploys(deploys: &mut DeployCollection, current_instant: Timestamp) -> usize {
+pub(super) fn prune_deploys(
+    deploys: &mut HashMap<DeployHash, DeployHeader>,
+    current_instant: Timestamp,
+) -> usize {
     let initial_len = deploys.len();
     deploys.retain(|_hash, header| !header.expired(current_instant));
     initial_len - deploys.len()
 }
+
+/// Prunes expired deploy information from an individual pending deploy collection, returns the
+/// total deploys pruned
+pub(super) fn prune_pending_deploys(
+    deploys: &mut HashMap<DeployHash, DeployType>,
+    current_instant: Timestamp,
+) -> usize {
+    let initial_len = deploys.len();
+    deploys.retain(|_hash, wrapper| match wrapper {
+        DeployType::Transfer(header) => !header.expired(current_instant),
+        DeployType::Wasm(header) => !header.expired(current_instant),
+    });
```

Comment:
I think that if there was a `.header()` method on `DeployType`, this could be reduced to
```rust
deploys.retain(|_hash, wrapper| !wrapper.header().expired(current_instant));
```

Reply:
Oh, you already have this method. We just need to use it.